tests/http/RequestTest.php

```diff
@@ -46,7 +46,7 @@ public function testBasic()
 
         $request = new Request([
             'method' => 'POST',
-            'uri' => new Uri('', ['query' => 'a=b&b=c']),
+            'uri' => new Uri('', ['query' => 'a=b&b=c&r.n=a&r-n=a&o[]=1&o[]=2']),
```

Comment:
What's the `o[]` use case? why don't we just passing `o=1,2`, it should be samper and more straightforward.

Reply:
Just for compatible, 'cause that's the default behavior of `parse_str`

```
>>> $query = 'o[]=1&o[]=2';
=> "o[]=1&o[]=2"
>>> parse_str($query, $params)
=> null
>>> $params
=> [
     "o" => [
       "1",
       "2",
     ],
   ]
```

Reply:
Oh, i know.

But this PR is already leads the incompatible with `parse_str`,  so i didn't think we must compatible with `parse_str`.

Instead, a sampler and more straightforward way is perfered, i think.